Source/Particles/Collision/BackgroundMCCCollision.cpp

```diff
@@ -0,0 +1,385 @@
+    // included the mass of the secondary species of that interaction
+    // will be used. If no neutral mass is specified and ionization is not
+    // included the mass of the colliding species will be used
+    m_background_mass = -1;
+    pp.query("background_mass", m_background_mass);
+
+    // query for a list of collision processes
+    // these could be elastic, excitation, charge_exchange, back, etc.
+    amrex::Vector<std::string> scattering_process_names;
+    pp.queryarr("scattering_processes", scattering_process_names);
+
+    // create a vector of MCCProcess objects from each scattering
+    // process name
+    for (auto scattering_process : scattering_process_names) {
+        std::string kw_cross_section = scattering_process + "_cross_section";
+        std::string cross_section_file;
+        pp.query(kw_cross_section.c_str(), cross_section_file);
+
+        amrex::Real energy = 0.0;
+        // if the scattering process is excitation or ionization get the
+        // energy associated with that process
+        if (scattering_process.find("excitation") != std::string::npos ||
+            scattering_process.find("ionization") != std::string::npos) {
+            std::string kw_energy = scattering_process + "_energy";
+            pp.get(kw_energy.c_str(), energy);
+        }
+
+        auto process = new MCCProcess(scattering_process, cross_section_file, energy);
+
+        // if the scattering process is ionization get the secondary species
+        // only one ionization process is supported, the vector
+        // m_ionization_processes is only used to make it simple to calculate
+        // the maximum collision frequency with the same function used for
+        // particle conserving processes
+        if (process->m_type == MCCProcessType::IONIZATION) {
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(!ionization_flag,
+                                             "Background MCC only supports a single ionization process");
+            ionization_flag = true;
+
+            std::string secondary_species;
+            pp.get("ionization_species", secondary_species);
+            m_species_names.push_back(secondary_species);
+
+            m_ionization_processes.push_back(process);
+            amrex::Gpu::synchronize();
+        } else {
+            addProcess(process);
+        }
+    }
+}
+
+void
+BackgroundMCCCollision::addProcess(MCCProcess* process)
+{
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(process != nullptr,
+                                     "Cannot add a NULL MCC process type");
+
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(process->m_type != MCCProcessType::INVALID,
+                                     "Cannot add an unknown MCC process type");
+
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(process->m_type != MCCProcessType::IONIZATION,
+                                     "Addition of ionization process requires special handling");
+
+    m_scattering_processes.push_back(process);
+
+    amrex::Gpu::synchronize();
+}
```

Comment:
It'll probably be cleaner to just inline this function. Some of the assertions can be obviated, and the sync can be factored out of the if (and the loop)

Reply:
Done